lib/rmt/downloader.rb

```diff
@@ -85,36 +85,26 @@ def verify_checksum(filename, checksum_type, checksum_value)
 
   def make_request(remote_file, local_file, request_fiber, checksum_type = nil, checksum_value = nil)
     uri = URI.join(@repository_url, remote_file)
-    uri.query = @auth_token if @auth_token
+    uri.query = @auth_token if (@auth_token && uri.scheme != 'file')
+
+    if (URI(uri).scheme == 'file' && !File.exist?(uri.path))
```

Comment:
Rubocop doesn't complain!!! :scream:

Reply:
I think it's the third time you say that, but... so what? Just because rubocop doesn't complain doesn't mean it doesn't look ugly...

Reply:
@lagartoflojo The whole point of using Rubocop is so that issues like that don't pop up during the review (because fixing some minor style issue manually every time is a huge annoyance and waste of time -- that should happen automagically) and also so that all the rules are uniform and codified (other reviewers don't complain about this particular issue).

So it either has to be a Rubocop rule, or you'll have to accept parens :-)

P.S. On a side note, many things are ugly. JavaScript is ugly, let's stop using JavaScript. :-)

Reply:
But Hernan does! :laughing:

Reply:
Y'all need some Perl code :grimacing:

Reply:
> So it either has to be a Rubocop rule, or you'll have to accept parens :-)

Hm, not sure if they even is such a rule, but if it was, I would vote against forcing one or the other option. Because sometimes it helps with readability and a pair of parens can save you, future-you or any other developer a few seconds of thinking. And sometimes the logic of an "complex" if-statement is so obvious that the parens itself become a distraction. How you perceive this, is different per developer and can always lead to discussions.
So I'm fine with or without ... In this case, _I_ don't think there is a need for them, so I removed them. Which I probably should not have done to let you guys fight it out :stuck_out_tongue_winking_eye:

Reply:
@thutterer There's a rule that fixes simple issues like `puts 1 if (1)`.

But yeah -- sometimes you need the parens. That's what I'm saying -- if Rubocop doesn't complain, then it's not an issue.

P.S. Does that mean that I can demand that you add the parens when I'm doing the reviews? Because I like parens! :grimacing: